### ISSUE TITLE
New Rule: CodePipeline Namespace validation

### ISSUE THREAD

**Issue by whoDoneItAgain** [aws-cloudformation/cfn-lint#4015]
### Is this feature request related to a new rule or cfn-lint capabilities?

rules

### Describe the feature you'd like to request

Validate that CodePipeline namespace references are valid

### Describe the solution you'd like

```
                  ParameterOverrides: !Sub
                    - |
                      {
                        "LambdaS3Key": "#{file-gw-r53-handler.LAMBDA_S3_KEY}",
                        "LambdaS3ObjectVersion": "#{file-gw-r53-handler.LAMBDA_S3_VERSION}"
                      }
```

this snippet references a namespace that doesn't exist any more in the pipeline. cfn-lint didnt catch it.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
